src/Package/Impl/DataInspect/Office/CsvAppFileIO.cs

```diff
@@ -28,68 +30,67 @@ internal static class CsvAppFileIO {
         private const string _variableNameReplacement = "variable";
         private static int _busy;
 
-        public static async Task OpenDataCsvApp(IREvaluationResultInfo result) {
-            await VsAppShell.Current.SwitchToMainThreadAsync();
+        public static async Task OpenDataCsvApp(IREvaluationResultInfo result, IApplicationShell appShell, IFileSystem fileSystem, IProcessServices processServices) {
+            await appShell.SwitchToMainThreadAsync();
 
             if (Interlocked.Exchange(ref _busy, 1) > 0) {
                 return;
             }
 
-            var workflowProvider = VsAppShell.Current.ExportProvider.GetExportedValue<IRInteractiveWorkflowProvider>();
-            var workflow = workflowProvider.GetOrCreate();
+            var workflow = appShell.ExportProvider.GetExportedValue<IRInteractiveWorkflowProvider>().GetOrCreate();
             var session = workflow.RSession;
 
             var folder = GetTempCsvFilesFolder();
             if (!Directory.Exists(folder)) {
                 Directory.CreateDirectory(folder);
             }
 
+            var pss = appShell.ExportProvider.GetExportedValue<IProjectSystemServices>();
             var variableName = result.Name ?? _variableNameReplacement;
-            var csvFileName = MakeCsvFileName(variableName);
+            var csvFileName = MakeCsvFileName(appShell, pss, variableName);
 
-            var pss = VsAppShell.Current.ExportProvider.GetExportedValue<IProjectSystemServices>();
             var file = pss.GetUniqueFileName(folder, csvFileName, "csv", appendUnderscore: true);
 
             string currentStatusText;
-            var statusBar = VsAppShell.Current.GetGlobalService<IVsStatusbar>(typeof(SVsStatusbar));
+            var statusBar = appShell.GetGlobalService<IVsStatusbar>(typeof(SVsStatusbar));
```

Comment:
Hm... probably it is not part of this change, but since we have `IStatusBar`, probably we should use it instead. That will allow to move this code to the `Components` assembly.